doc/basics.md

```diff
@@ -1080,10 +1081,10 @@ auto cars = parser.iterate(cars_json);
 cout << cars.at_pointer("/0/tire_pressure/1") << endl; // Prints 39.9
 ```
 
-A JSON Path is a sequence of segments each starting with the '/' character. Within arrays, an integer
+A JSON Pointer path is a sequence of segments each starting with the '/' character. Within arrays, an integer
 index allows you to select the indexed node. Within objects, the string value of the key allows you to
 select the value. If your keys contain the characters '/' or '~', they must be escaped as '~1' and
```

Comment:
> If your keys contain the characters '/' or '~'

I understand that / is used as a separator, hence, needs escaping, but I wonder why ~ needs escaping.

Curious, do we allow unicode characters?

Reply:
This comes from the json pointer spec: https://datatracker.ietf.org/doc/html/rfc6901#section-4

Since we use '~1' to escape '/', there has to be a way for the user to specify "~1" as part of a valid string. Imagine that we want to specify the following path "/someField~1/...", without having the "~1" being replaced by "/". For this to happen, we need to escape the "~" as "~0" and then the string can be represented as "/someField~01/"/

@lemire can you clarify regarding unicode character support? (I see it in the JSON pointer spec, but you have a lot more context here)

Reply:
> Since we use '~1' to escape '/', there has to be a way for the user to specify...

Got it. Thanks.

Reply:
Oh, the markdown text for the message got super weird from using `~`. Just updated the comment and it seems that the strikethrough effect is now gone from the markdown format.

Reply:
@lemire can you chime in here regarding @mbasmanova question on unicode characters?

Reply:
@FranciscoThiesen @mbasmanova 

Regarding unicode support, I suppose you mean within the JSON Pointer string. With JSON Pointer, the only place it makes sense to have non-ASCII characters is when matching keys. The current implementation assumes that the keys are ASCII which is overwhelmingly the case with JSON. It is not a fundamental limitation but this is done for speed.

I am not sure if it is currently documented because I don't think anyone ever asked the question. At a minimum, it should be documented.

Reply:
(Regarding my comment, I know we mean JSON Path, but I was answering regarding what exists currently, prior to this PR.)

Reply:
@FranciscoThiesen @mbasmanova My statement was incorrect. We will support unicode (UTF-8) just fine, but not if there are escaped Unicode sequences.

Reply:
So the key `"école"` would be matched. No problem. But only if it appears byte-by-byte identical in the path and in the JSON document.

It has to be UTF-8 throughout. We don't do normalization. We don't escape.

Reply:
@lemire Goti it. Thank you for clarifying. If not too much trouble, it would be nice to document this.